fs/reader/reader.go

```diff
@@ -201,13 +203,18 @@ func (gr *reader) isClosed() bool {
 }
 
 type file struct {
-	id uint32
-	fr metadata.File
-	gr *reader
+	id       uint32
+	fr       metadata.File
+	gr       *reader
+	verified atomic.Bool
+	lock     sync.Mutex
```

Comment:
I thought about this, but opted not to right now because `sync.Once` doesn't have any way to retry. A temporary network issue would continuously fail, even if the container workload redrives the request after receiving `EIO`.

It might be worth creating a custom retriable `sync.Once` with rate limiting. What do you think?

Reply:
I was imaging a retriable `sync.Once` where on failure, it's locked out until the retry interval is up. Any call during that lock out would return immediately without doing any work. On success, it would be marked completed and would always return immediately.

My only concern with not rate limiting is that this verification has a network call and is driven by filesystem operations which are probably not design around expected network conditions. If there is a temporary failure which causes failed validation and the container does retries (which is already a big condition), then I'm worried that retrying the network request as fast as possible causes even more network issues which cause the container to die anyway. If we did have a lockout, the container might die anyway waiting for the next retry period.